### ISSUE TITLE
Recolorize NaN error after alignment 

### ISSUE THREAD

**Issue by C-G-L** [hiweller/recolorize#5]
Hi Hannah,

I am using Recolorize to analyze Heliconius wings and I am following the recolorize -> patternize pipeline using the recolorize_examples. I have created a similar file structure as the one used in the recolorize examples files. The images have had the background removed using rembg and are png's. The images are available here:  
[erato_lativitta.zip](https://github.com/hiweller/recolorize/files/11499685/erato_lativitta.zip)

The images align well, and I am able to generate reconstructions in the R-plot space. However, when I go to plot them all using the lapply function, I get an error: error in rgb color intensity nan not in (0,1). And upon further inspection, there are NaN's in the sample_raster_data attribute. Like so. 

Have you encountered this error before or have any suggestions for how to proceed? 

Thank you,

Christopher

![Screenshot from 2023-05-16 09-55-39](https://github.com/hiweller/recolorize/assets/73145669/eff02a0f-9f5b-4bbc-9c38-9b057769e93d)
![Screenshot from 2023-05-16 09-56-32](https://github.com/hiweller/recolorize/assets/73145669/a1b292fa-f36d-42ca-a6e4-1671df7755c8)
![Screenshot from 2023-05-16 10-58-10](https://github.com/hiweller/recolorize/assets/73145669/1efcd165-dbf1-4535-b4b3-97b02d25191a)

**Comment by hiweller** [comment]
Hi Christopher,

Thanks for bringing this bug to my attention! I was able to reproduce the same problem on my computer using your attached files. It seems that sometimes in the alignment step, patternize ends up generating 'NaN' values, and the `brick_to_array` function in recolorize doesn't correctly convert those into 0's to represent background. It looks like this is largely the result of rotating images to be aligned (you can see that the NaN values are all the result of rotating a rectangle):

```{r}
# load raster library
library(raster)

# read in raster list from erato_lativitta
rasterlist <- readRDS("erato_lativitta/fws_imageList_aligned.rds")

# generate plots of where the NaN values are (in green):
layout(matrix(1:10, nrow = 2))
par(mar = c(2, 3, 3, 1))
b <- lapply(1:length(rasterlist), \(x) plot(merge(is.nan(rasterlist[[x]])),
                                            main = names(rasterlist)[x]))
```

![nan_maps](https://github.com/hiweller/recolorize/assets/12073558/23207846-1a08-4328-bbd3-8c77e047245a)

If you look at the [brick_to_array](https://github.com/hiweller/recolorize/blob/master/R/brick_to_array.R) function in recolorize, you'll see that the only way we identify background pixels is by looking for pixels with an alpha value of 0. So by default, the NaN values get treated as being in the region of interest, carry through to the converted array as if they're regular RGB pixels, and then confuse the downstream functions that are expecting RGB values in the 0-1 range.

You could in theory fix that by just converting all the NaN and NA values in each raster to a 0 before running them through `brick_to_array`:

```{r}
for (i in 1:length(raster_list)) {
    r <- raster_list[[i]]
    r[is.na(r)] <- 0
    raster_list[[i]] <- r
}
```

...but, that feels silly when I could fix this problem entirely by just adding a single line (`r[is.na(r)] <- 0`) to the `brick_to_array` function, so [that's what I did](https://github.com/hiweller/recolorize/commit/5b1b559c4ca1b91585eeabb9da5f49deda0959a3). This converts all NaN and NA values to 0 before looking for pixels with an alpha of 0 to treat as background. This seems to fix the problem, because after updating recolorize the following code runs with no problem:

```{r}
a_list <- lapply(rasterlist, brick_to_array)
lapply(a_list, plotImageArray)
```

![image-20230517153110982](https://github.com/hiweller/recolorize/assets/12073558/031a6142-1600-474e-8f26-1f1fdfd106bb)

The obvious caveat here is that if you want your NaN/NA values to represent something other than background, this will eliminate them, so it's a pretty patternize-specific solution. That said, recolorize doesn't really have a way of handling NA/NaN values besides ignoring them, so in this case it seems appropriate.

## TL;DR if you're having the same problem

Update recolorize from source by running `devtools::install_github("hiweller/recolorize")`, restart your R session, and try converting from raster bricks to arrays again.

Let me know if this resolves the issue for you!
